packages/origin-247-certificate/src/evented-certificate/repositories/CertificateEvent/CertificateEvent.entity.ts

```diff
@@ -0,0 +1,32 @@
+import { CreateDateColumn, Entity, PrimaryGeneratedColumn, Column } from 'typeorm';
+import { IIssueCommand, ITransferCommand, IClaimCommand } from '../../../types';
+import { CertificateEventType } from '../../events/Certificate.events';
+
+export const tableName = 'certificate_event';
+
+@Entity(tableName)
+export class CertificateEventEntity {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column({ nullable: false })
+    internalCertificateId: number;
+
+    @Column({ nullable: true })
+    blockchainCertificateId: number | null;
```

Comment:
and this is not required

Reply:
What do you mean that it's not required?

This is the field that I have in mind for certificateId in case of persisted events. In case of internal certificates it would be null and in case of persisted events ( after synchronization ) it would have the actual certificateId.

Reply:
IssuePersistedEvent will contain blockchainCertificateId in it's payload. You don't need that on each event. And events cannot be modified by design, because they are events, so it doesn't make sense anyway.

Then, when you parse events into read model you can have this blockchainId there, if you want.